hash.go

```diff
@@ -102,6 +106,179 @@ func RotateSequence(sequence string) string {
 	return sequence
 }
 
+/******************************************************************************
+Dec, 2, 2020
+
+Seqhash stuff starts here.
+
+There is a big problem with current sequence databases - they all use different
+identifiers and accession numbers. This means cross-referencing databases is
+a complicated exercise, especially as the quantity of databases increases, or if
+you need to compare "wild" DNA sequences.
+
+Seqhash is a simple algorithm to produce consistent identifiers for any genetic sequence. The
+basic premise of the Seqhash algorithm is to hash sequences with the hash being a robust
+cross-database identifier. Sequences themselves shouldn't be used as a database index
+(often, they're too big), so a hash based off of a sequence is the next best thing.
+
+Usability wise, you should be able to Seqhash any rotation of a sequence in any direction and
+get a consistent hash.
+
+The Seqhash algorithm makes several opinionated design choices, primarily to make working
+with Seqhashes more consistent and nice. The Seqhash algorithm only uses a single hash function,
+Blake3, and only operates on DNA, RNA, and Protein sequences. These identifiers will be seen
```

Comment:
what is a single hash function/Blake3?

Reply:
A single hash function is a hash function in contrast to hash functions, ie, singular instead of plural. 

https://github.com/BLAKE3-team/BLAKE3 BLAKE3 is a fast cryptographic hash function